e2e/terraform/Makefile

```diff
@@ -0,0 +1,10 @@
+NOMAD_SHA := $(shell git rev-parse HEAD)
+
+dev-cluster:
+	terraform apply -auto-approve -var-file=terraform.tfvars.dev
```

Comment:
It looks like the intent here is to use `terrafrom.tfvars.dev` as a copy of `terraform.tfvars`. We should probably call that out in the README and add it to the `.gitignore`. Maybe make it a dependency of the `dev-cluster` target, too?

Reply:
Oops - I pushed the dev file now - it's simply 3 server, 2 linux clients, and no windows; as that's I suspect suffices most dev manual testing use cases.